src/Service/PriceRuleService.php

```diff
@@ -394,7 +394,7 @@ public function applyPriceRuleForCarrier(Cart $cart, $priceRulesIds, $shopId)
                 return false;
             }
             $priceRule = new DPDPriceRule($priceRuleId, null, $shopId);
-
+            $priceRule->price += $this->getAdditionalShippingCosts($cart);
```

Comment:
Wouldn't really say it's a PriceRule and `applyPriceRuleForCarrier` isn't the place this should be added. 
Maybe shippingPriceCalculator that handles priceRules, additionalShippingCosts and other stuff that may appear?

Reply:
@tomjas1997  do u mean to create separated sevice to add additional shipping costs for carrier?

Reply:
PriceRule has it's own tab inside PS, "additional fees" are not included in that. 

I mean to create shippingPriceCalculator service and add this method there and call PriceRuleService as well, this way there's only one place that needs to be called and you wont be putting additional logic into where it's not supposed to be in

Reply:
Okey got u. I will make a separated service for shipping price calculation.